src/main/java/net/dv8tion/jda/internal/interactions/command/SlashCommandReference.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.interactions.command;
+
+import net.dv8tion.jda.api.interactions.commands.ICommandReference;
+
+public class SlashCommandReference implements ICommandReference
+{
+    private final long id;
+    private final String name;
+
+    public SlashCommandReference(String name, long id)
+    {
+        this.name = name;
+        this.id = id;
+    }
+
+    @Override
+    public long getIdLong()
+    {
+        return id;
+    }
+
+    @Override
+    public String getName()
+    {
+        return name;
+    }
+
+    @Override
+    public String toString()
+    {
+        return String.format("SlashCommandReference: name=%s, id=%d", name, id);
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        SlashCommandReference that = (SlashCommandReference) o;
+        return id == that.id;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return (int) (id ^ (id >>> 32));
```

Comment:
Should use `Long.hashCode` but also check that the name is the same. A subcommand mention uses the same id but has a different name, which should be refelected by equals and hashcode.